compiler/shared/main/scala/mlscript/compiler/optimizer/TailRecOpt.scala

```diff
@@ -66,37 +77,57 @@ class TailRecOpt(fnUid: FreshInt, classUid: FreshInt, tag: FreshInt):
       if names.contains(nme) then Some(nme)
       else None
 
-  // would prefer to have this inside discoverOptimizableCalls, but this makes scala think it's not tail recursive
+  // would prefer to have this inside discoverOptCalls, but scala does not support partially tail recursive functions directly
```

Comment:
I think you could just add `private val discoverOptCallsIndirect = discoverOptCalls` and call that instead.

Reply:
This doesn't seem to work due to the `implicit` parameters :/

Either way, I don't think it is a big deal so I'll resolve this for now. If you disagree you can re-open this comment.